R/report.R

```diff
@@ -142,11 +141,22 @@ get_results <- function(report, unnest = FALSE) {
 #'
 #' @param report Report object that stores validation results. See \link{get_results}.
 #' @param file_name Name of the resulting file (including extension).
-#' @param method Function that should be used to save results table (write.csv default).
+#' @param method Function that should be used to save results table (write.csv default)
+#' The function passed to \code{method} should have 'x' and 'file' arguments. Functions with
+#' different arguments can be passed by creating a wrapper function for it.
+#' See example save_results_methods.
 #' @param ... Remaining parameters passed to \code{method}.
 #' @export
 save_results <- function(report, file_name = "results.csv", method = utils::write.csv, ...) {
-  report$save_results(file_name, method, ...)
+  tryCatch({
+    report$save_results(file_name, method, ...)
+  },
+  error = function(error) {
+    rlang::abort(
+      "Method is not supported. Method must have 'x' and 'file' arguments.
+      Please create a wrapper function to use un-supported methods."
+    )
+  })
```

Comment:
1. `save_result` is just a facade function, the logic is in the R6 class method, thus it makes little sense to put the error handling here.
2. Saving can fail not only due to the problem with the method parameter names. The error message should a) include the original error message, so it is possible to debug the problem, b) only suggest that the problem might be in the names of parameters.